turbopuffer/namespace.py

```diff
@@ -277,6 +277,13 @@ def upsert(self,
         elif isinstance(data, VectorColumns):
             payload = {**data.__dict__}
 
+            # Convert List[float] vectors to base64-encoded strings, if enabled.
+            if tpuf.upsert_vectors_as_base64 and payload["vectors"] is not None:
+                payload["vectors"] = [
+                    b64encode_vector(vector) if isinstance(vector, list) else vector
+                    for vector in payload["vectors"]
+                ]
```

Comment:
Would it be any more efficient to encode these vectors into a single flat string?

```suggestion 
                payload["vectors"] = "".join(
                    b64encode_vector(vector) if isinstance(vector, list) else vector
                    for vector in payload["vectors"]
                )
```

Reply:
If we did that then we would need to handle the NULL (delete vector) case, which would introduce enough complexity that it's probably not worth it.

Reply:
Ahh, yes, of course. I think probably you could encode null as a vector of all NaNs, but last I heard we’d like to get rid of that. Agree on it not being worth it given that wrinkle.